### PR TITLE
Continue replaying checkpoint on duckdb restart if we see a checkpoint entry in the WAL

### PR DESCRIPTION
**Pull request by Vegetable26**
The presence of this entry means DuckDB was attempting to perform a checkpoint when it crashed last time. Let's just finish up the checkpoint on restart.

### CODE REVIEW

File: src/storage/storage_manager.cpp [hunk 1]
```diff
@@ -146,13 +146,16 @@ void SingleFileStorageManager::LoadDatabase() {
...
 			// replay the WAL
-			truncate_wal = WriteAheadLog::Replay(db, wal_path);
+			replay_status = WriteAheadLog::Replay(db, wal_path);
 		}
 	}
 	// initialize the WAL file
 	if (!read_only) {
 		wal = make_uniq<WriteAheadLog>(db, wal_path);
-		if (truncate_wal) {
+		if (replay_status.should_checkpoint) {
```
Comment (Tishj) [inline src/storage/storage_manager.cpp:155]: I might be reading this wrong, but can't `should_checkpoint` be uninitialized here?
No default value for the variables is provided.

Comment (Vegetable26) [reply src/storage/storage_manager.cpp:155]: Oh sorry - I think I forgot to initialize after removing the default values. Thanks for the catch!

**Comment by Vegetable26** [comment]
It's failing some failure-injection tests that inject failures on checkpoint. Will take a look at them tomorrow - easiest fix will be to disable the failure-injection on restart.

**Comment by Mytherin** [comment]
Closing for now - feel free to re-open if you want to continue to work on this

**Pull request closed without merging**